src/bundler/bundle_v2.zig

```diff
@@ -7546,6 +7548,90 @@ pub const LinkerContext = struct {
         return hasher.digest();
     }
 
+    pub fn validateTLA(c: *LinkerContext, source_index: Index.Int) js_ast.TlaCheck {
+        var result_tla_check: *js_ast.TlaCheck = &c.parse_graph.ast.items(.tla_check).ptr[source_index];
+        const result = c.parse_graph.ast.get(source_index);
+
+        if (result_tla_check.depth == 0) {
+            result_tla_check.depth = 1;
+            if (result.top_level_await_keyword.len > 0) {
+                result_tla_check.parent = source_index;
+            }
+
+            for (result.import_records.slice(), 0..) |record, import_record_index| {
+                if (Index.isValid(record.source_index) and (record.kind == .require or record.kind == .stmt)) {
+                    const parent = c.validateTLA(record.source_index.get());
+                    if (Index.isInvalid(Index.init(parent.parent))) {
+                        continue;
+                    }
+
+                    // Follow any import chains
+                    if (record.kind == .stmt and (Index.isInvalid(Index.init(result.tla_check.parent)) or parent.depth < result.tla_check.depth)) {
+                        result_tla_check.depth = parent.depth + 1;
+                        result_tla_check.parent = record.source_index.get();
+                        result_tla_check.import_record_index = @intCast(import_record_index);
+                        continue;
+                    }
+
+                    // Require of a top-level await chain is forbidden
+                    if (record.kind == .require) {
+                        var notes = std.ArrayList(Logger.Data).init(c.allocator);
+                        var tla_pretty_path: string = "";
+                        var other_source_index = record.source_index.get();
+
+                        // Build up a chain of notes for all of the imports
+                        while (true) {
+                            const parent_result = c.parse_graph.ast.get(other_source_index);
```

Comment:
Do we need to call .get() here? This loads data from many different arrays.

Reply:
only top_level_await_keywords and tla_check are referenced?

Reply:
https://github.com/oven-sh/bun/pull/14218/commits/b6add9c5314fddbdecff3f925a6bd09fd6517e1c